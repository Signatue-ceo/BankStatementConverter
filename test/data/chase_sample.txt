CHASE BANK
Account Statement
Account Number: ****1234
Statement Period: 01/01/2024 to 01/31/2024

Opening Balance: $5,000.00
Closing Balance: $4,850.00

Date        Description                    Amount      Balance
01/02/2024  GROCERY STORE PURCHASE        -$45.67     $4,954.33
01/05/2024  GAS STATION                    -$35.00     $4,919.33
01/08/2024  SALARY DEPOSIT                +$2,500.00  $7,419.33
01/12/2024  RESTAURANT DINNER             -$85.50     $7,333.83
01/15/2024  ONLINE SHOPPING               -$120.00    $7,213.83
01/18/2024  UTILITY BILL PAYMENT          -$150.00    $7,063.83
01/22/2024  ATM WITHDRAWAL                -$200.00    $6,863.83
01/25/2024  PHARMACY                      -$25.30     $6,838.53
01/28/2024  MOVIE TICKETS                 -$45.00     $6,793.53
01/30/2024  COFFEE SHOP                   -$12.50     $6,781.03
01/31/2024  BANK FEE                      -$15.00     $6,766.03
