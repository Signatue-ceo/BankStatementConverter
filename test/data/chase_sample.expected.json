{
  "id": "mock-id",
  "bankName": "CHASE BANK",
  "accountNumber": "****1234",
  "statementPeriod": {
    "start": "2024-01-01T05:00:00.000Z",
    "end": "2024-01-31T05:00:00.000Z"
  },
  "openingBalance": 5000,
  "closingBalance": 4850,
  "transactions": [
    {
      "id": "mock-id",
      "date": "2024-01-02T00:00:00.000Z",
      "description": "GROCERY STORE PURCHASE",
      "amount": 45.67,
      "type": "debit",
      "balance": 4954.33,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-01-05T00:00:00.000Z",
      "description": "GAS STATION",
      "amount": 35,
      "type": "debit",
      "balance": 4919.33,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-01-08T00:00:00.000Z",
      "description": "SALARY DEPOSIT",
      "amount": 2500,
      "type": "credit",
      "balance": 7419.33,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-01-12T00:00:00.000Z",
      "description": "RESTAURANT DINNER",
      "amount": 85.50,
      "type": "debit",
      "balance": 7333.83,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-01-15T00:00:00.000Z",
      "description": "ONLINE SHOPPING",
      "amount": 120,
      "type": "debit",
      "balance": 7213.83,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-01-18T00:00:00.000Z",
      "description": "UTILITY BILL PAYMENT",
      "amount": 150,
      "type": "debit",
      "balance": 7063.83,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-01-22T00:00:00.000Z",
      "description": "ATM WITHDRAWAL",
      "amount": 200,
      "type": "debit",
      "balance": 6863.83,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-01-25T00:00:00.000Z",
      "description": "PHARMACY",
      "amount": 25.30,
      "type": "debit",
      "balance": 6838.53,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-01-28T00:00:00.000Z",
      "description": "MOVIE TICKETS",
      "amount": 45,
      "type": "debit",
      "balance": 6793.53,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-01-30T00:00:00.000Z",
      "description": "COFFEE SHOP",
      "amount": 12.50,
      "type": "debit",
      "balance": 6781.03,
      "reference": null
    },
    {
      "id": "mock-id",
      "date": "2024-01-31T05:00:00.000Z",
      "description": "BANK FEE",
      "amount": 15,
      "type": "debit",
      "balance": 6766.03,
      "reference": null
    }
  ],
  "metadata": {
    "processingTime": 0,
    "confidence": 0,
    "sourceFormat": "pdf",
    "totalTransactions": 11
  }
}